sound/soc/sof/stream-ipc.c

```diff
@@ -34,13 +34,28 @@ int sof_ipc_msg_data(struct snd_sof_dev *sdev,
 		snd_sof_dsp_mailbox_read(sdev, sdev->dsp_box.offset, p, sz);
 	} else {
 		struct snd_pcm_substream *substream = sps->substream;
-		struct sof_stream *stream = substream->runtime->private_data;
+		size_t posn_offset;
+
+		if (substream) {
+			struct sof_stream *stream = substream->runtime->private_data;
+
+			/* The stream might already be closed */
+			if (!stream)
+				return -ESTRPIPE;
+
+			posn_offset = stream->posn_offset;
+		} else {
```

Comment:
This `else` case  will never happen as the higher level check at line 33 will direct to `snd_sof_dsp_mailbox_read()` if sps->substream is NULL.